packages/react/src/components/cookieConsentCore/cookieConsentCore.js

```diff
@@ -0,0 +1,670 @@
+    this.#SETTINGS_PAGE_SELECTOR = settingsPageSelector;
+
+    window.hds = window.hds || {};
+    window.hds.cookieConsent = this;
+
+    /**
+     * Updates the shadow DOM checkboxes based on the consented group names.
+     *
+     * @param {Array<string>} consentedGroupNames - The array of consented group names.
+     * @param {Array<string>} formReference - The reference to the form element.
+     */
+    const shadowDomUpdateCallback = (consentedGroupNames, formReference) => {
+      if (formReference) {
+        const formCheckboxes = formReference.querySelectorAll('input');
+        formCheckboxes.forEach((check) => {
+          // eslint-disable-next-line no-param-reassign
+          check.checked = consentedGroupNames.includes(check.dataset.group);
+        });
+      }
+    };
+
+    this.#COOKIE_HANDLER = new CookieHandler({ shadowDomUpdateCallback, siteSettingsObj, lang: language });
+    this.#MONITOR = new MonitorAndCleanBrowserStorages();
+  }
+
+  /**
+   * Creates and inits a new instance of the CookieConsent class.
+   * @param {Object} siteSettingsParam - The URL to the JSON file with site settings or contents of the site settings as an object.
+   * @param {Object} options - The options for configuring the CookieConsent instance.
+   * @param {string} [options.language='en'] - The page language.
+   * @param {string} [options.theme='bus'] - The theme for the banner.
+   * @param {string} [options.targetSelector='body'] - The selector for where to inject the banner.
+   * @param {string} [options.spacerParentSelector='body'] - The selector for where to inject the spacer.
+   * @param {string} [options.pageContentSelector='body'] - The selector for where to add scroll-margin-bottom.
+   * @param {boolean|string} [options.submitEvent=false] - If a string, do not reload the page, but submit the string as an event after consent.
+   * @param {string} [options.settingsPageSelector=null] - If this string is set and a matching element is found on the page, show cookie settings in a page replacing the matched element.
+   * @return {Promise<CookieConsentCore>} A promise that resolves to a new instance of the CookieConsent class.
+   * @throws {Error} Throws an error if the siteSettingsParam is not a string or an object.
+   * @throws {Error} Throws an error if siteSettingsParam is an URL string and the fetch fails.
+   * @throws {Error} Throws an error if siteSettingsParam is an URL string and the JSON parsing fails.
+   */
+  static async create(siteSettingsParam, options) {
+    let instance;
+    if (!siteSettingsParam && (typeof siteSettingsParam !== 'string' || typeof siteSettingsParam !== 'object')) {
+      throw new Error(
+        'Cookie consent: siteSettingsParam is required, it should be an URL string or an siteSettings object.',
+      );
+    }
+    if (typeof siteSettingsParam === 'string') {
+      // Fetch the site settings JSON file
+      const siteSettingsRaw = await fetch(siteSettingsParam).then((response) => {
```

Comment:
I did not find this easier to read at all, I had to read it multiple times to understand what's going on. :D

It may have been due to the indentation mistake on the first row, or the missing color highlighting, but this was way harder to read for me. :D 

I do understand that this may be different for each developer and I'm probably blind for my own idiotic coding style. (And I do admit that my code on this part is in no way pretty)

Maybe we can find a better way to handle this.